File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/view/WindowBuilder.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.view");
Clazz.load (null, "pl.wroc.pwr.student.acteditor.view.WindowBuilder", ["pl.wroc.pwr.student.acteditor.model.DocumentRegistry", "pl.wroc.pwr.student.acteditor.util.Utils", "pl.wroc.pwr.student.acteditor.view.SubWindow"], function () {
c$ = Clazz.decorateAsClass (function () {
this.display = null;
this.registry = null;
Clazz.instantialize (this, arguments);
}, pl.wroc.pwr.student.acteditor.view, "WindowBuilder");
Clazz.makeConstructor (c$, 
function (display) {
this.display = display;
this.registry = pl.wroc.pwr.student.acteditor.model.DocumentRegistry.getRegistry ();
}, "$wt.widgets.Display");
Clazz.defineMethod (c$, "openWindow", 
function (choice) {
 new pl.wroc.pwr.student.acteditor.view.SubWindow (this.display, pl.wroc.pwr.student.acteditor.util.Utils.map (this.registry.get (choice)));
}, "~N");
});

File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/view/WindowBuilder.java
package pl.wroc.pwr.student.acteditor.view;

import org.eclipse.swt.widgets.Display;

import pl.wroc.pwr.student.acteditor.model.DocumentRegistry;
import pl.wroc.pwr.student.acteditor.util.Utils;

/**
 * Zapewnia otwarcie właściwego okna edycji dokumentu.
 * 
 * @author Mateusz
 * 
 */
public class WindowBuilder {
	private Display display;
	private DocumentRegistry registry;

	/**
	 * Tworzy obiekt typu WindowBuilder.
	 * 
	 * @param display
	 *          Obecny watek wyswietlajacy
	 */
	public WindowBuilder(Display display) {
		this.display = display;
		this.registry = DocumentRegistry.getRegistry();
	}

	/**
	 * Otwiera wskazane okno.
	 * 
	 * @param choice
	 *          Identyfikator okna.
	 */
	public void openWindow(int choice) {
		new SubWindow(display, Utils.map((String) registry.get(choice)));
	}
}
